semantics/src/main/kotlin/com/strumenta/kolasu/semantics/symbol/resolver/SymbolResolver.kt

```diff
@@ -0,0 +1,70 @@
+package com.strumenta.kolasu.semantics.symbol.resolver
+
+import com.strumenta.kolasu.model.Node
+import com.strumenta.kolasu.model.PossiblyNamed
+import com.strumenta.kolasu.model.ReferenceByName
+import com.strumenta.kolasu.model.children
+import com.strumenta.kolasu.model.kReferenceByNameType
+import com.strumenta.kolasu.model.nodeProperties
+import com.strumenta.kolasu.semantics.scope.provider.ScopeProvider
+import kotlin.reflect.KProperty1
+import kotlin.reflect.full.isSubtypeOf
+
+/**
+ * Symbol resolver instances can be used to resolve references within AST nodes
+ * (i.e. `ReferenceByName` instances). Internally, the symbol resolver uses a scope provider
+ * defining the language-specific scoping rules. The resolution process can be invoked for
+ * specific node properties or for all properties of a given node (possibly its children as well).
+ *
+ * Symbol resolution can be executed invoking one of the following methods:
+ * - `resolve(node, reference)`: to resolve a specific reference of the given node;
+ * - `resolve(node, withChildren)`: to resolve all references of the given node
```

Comment:
This parameter is now called `entireTree`

Reply:
Ops, sorry, you're right - will fix it.